### PR TITLE
EM-1308: Header Large Basic to Header Small (ESB)

### PR DESCRIPTION
**Pull request by mrfantasticwonders**
**Purpose:**

> As a designer or a front end developer, I'd like the pages that use the "header_large_basic" header partial to henceforth use the "header_small" partial (which might need to be created) so that going forward we have:
> - a defined set of headers
> - the same language for this header type
> - generic, reusable components when we want to use this header type again

This PR introduces the `header--small()` mixin and refactors the `header-hero()` mixin to use what was pulled from `header--large-basic` property found on the employer side.

It also introduces two new variables: `header-padding` and `header-width-normal` (per Susan).

For some reason or the other, the subnav() mixin is being included. @arelia Please let me know if you wish to exclude this since this is technically not a part of the story.

**JIRA:**
https://cb-content-enablement.atlassian.net/browse/EM-1308

**Changes:**
* Changes to setup
  * N/A

* Architectural changes
  * N/A

* Migrations
  * N/A
  
* Library changes
  * N/A

* Side effects
  * N/A

**Screenshots**
* Before
header-before

* After
header-after

**QA Links:**
http://web.employer-7.development.c66.me/

**How to Verify These Changes**
* Specific pages to visit
  * http://web.employer-7.development.c66.me/recruiting-solutions/display-advertising
  * http://web.employer-7.development.c66.me/recruiting-solutions/talent-driver
  * http://web.employer-7.development.c66.me/recruiting-solutions/source-and-screen
  * http://web.employer-7.development.c66.me/recruiting-solutions/hiring-events
  * http://web.employer-7.development.c66.me/employment-screening/drug-testing-services
  * http://web.employer-7.development.c66.me/employment-screening/electronic-i9-e-verify
  * http://web.employer-7.development.c66.me/recruiting-solutions/job-posting-and-candidate-profile-trends

* Steps to take
  * Make sure the header looks ok. It should have 70px padding top and bottom. It should have a max-width of 650px (per Susan). Beyond that, compare it to the prod versions of the links above. The text should be the same, it should respond the same as well as carry the same properties excluding the refactored padding and width.

* Responsive considerations
  * N/A

**Relevant PRs/Dependencies:**
https://github.com/cbdr/employer/pull/898

**Additional Information**

### CODE REVIEW

File: sass/directives/_flyout.scss [hunk 1]
```diff
@@ -87,3 +87,122 @@
     }
   }
 }
+
+// Use the mobile-subnav mixin for the subnav bars that appear on our subscription / product pages
+// Displays a single link with a flyout
+@mixin mobile-subnav-flyout() {
```
Comment (arelia) [inline sass/directives/_flyout.scss]: 👍

Comment (toastercup) [inline sass/directives/_flyout.scss]: As @MKwenhua mentioned, you don't need the `()` for mixins that lack params. It matches what we've been doing stylistically, and is one of the things that makes it ever-so-slightly easier to sight-read mixins.

File: sass/directives/_headers.scss [hunk 2]
```diff
@@ -8,6 +8,31 @@
     @include align-items(center);
   }
 
+  &__headline {
+    @include span-columns(12);
+    padding-right: 20px;
+    padding: $header-padding 0;
+    max-width: $header-width-normal;
```
Comment (arelia) [inline sass/directives/_headers.scss]: this should probably go on the header--large--left mixin. the centered text headers, like small biz, may have wider text

File: sass/directives/_flyout.scss [hunk 3]
```diff
@@ -87,3 +87,122 @@
...
+  }
+
+  .mobile-subnav__close {
+    display: none;
+    
+    &:after {
+      font-family: FontAwesome;
+      font-size: $top-bar-angle-size;
+      color: $base-link-color;
+      content: '\f107';
```
Comment (mrfantasticwonders) [inline sass/directives/_flyout.scss]: In the revised version in 1359, the new hover state for the initial link Susan requested includes unicode for the opposite angle.

**Comment by mrfantasticwonders** [comment]
@toastercup @arelia @MKwenhua Changes have been added to this PR if you want to re-review. I am going to send this off to Susan for design review.

File: sass/directives/_headers.scss [hunk 4]
```diff
@@ -172,3 +172,28 @@
     }
   }
 }
+
+@mixin header--small {
+  max-height: 300px;
```
Comment (arelia) [inline sass/directives/_headers.scss]: will you please make this a variable next to the other header height variables: https://github.com/cb-talent-development/employer-style-base/blob/master/sass/variables/_sizing.scss#L72

File: sass/directives/_headers.scss [hunk 5]
```diff
@@ -172,3 +172,28 @@
...
+
+  &__headline {
+    padding: $header-padding 0;
+    max-width: $header-width-normal;
+ 
+    h1 {
```
Comment (arelia) [inline sass/directives/_headers.scss]: these margins should probably be set in an existing or a new pattern from https://github.com/cb-talent-development/employer-style-base/blob/master/sass/directives/_headings.scss

File: sass/directives/_flyout.scss (hunk 3 above)
Comment (arelia) [inline sass/directives/_flyout.scss]: 👍

**Pull request merged**